### PR TITLE
[Merge] 8.0.x to develop

### PR DESCRIPTION
**Pull request by matthieu-rolland**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Merge 8.0.x to develop 
| Type?             | improvement
| Category?         | ME 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| Related PRs       | ~
| How to test?      | 
| Possible impacts? |

### CODE REVIEW

File: src/Adapter/Product/Combination/Repository/CombinationRepository.php [hunk 1]
```diff
@@ -259,4 +264,72 @@ public function getDefaultCombinationId(ProductId $productId): ?CombinationId
...
+
+        return $id ? $this->get(new CombinationId($id)) : null;
+    }
+
+    /**
+     * @param int[] $attributeIds
+     *
+     * @return CombinationId[]
+     */
+    public function getCombinationIdsByAttributes(ProductId $productId, array $attributeIds): array
```
Comment (jolelievre) [inline src/Adapter/Product/Combination/Repository/CombinationRepository.php:295]: This one is not used anymore, but we can clean it later

File: src/Core/Repository/ShopConstraintTrait.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Repository;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+trait ShopConstraintTrait
```
Comment (jolelievre) [inline src/Core/Repository/ShopConstraintTrait.php:34]: @zuk3975 the trait I mentioned will be be available on develop once this PR is merged

**Review by jolelievre — DISMISSED** [review]
Thanks @matthieu-rolland

**Comment by jolelievre** [comment]
Automatic campaigns are running here https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/3490649561

**Review by matks — DISMISSED** [review]

**Comment by zuk3975** [comment]
note build is failing

**Review by jolelievre — DISMISSED** [review]

**Comment by jolelievre** [comment]
Auto campaign running here https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/3497637685

**Comment by jolelievre** [comment]
The failing tests in auto campaign should be fixed by this PR https://github.com/PrestaShop/PrestaShop/pull/30326
I think we should still force merge this PR or the merge PR will keep growing week after week, and as soon as the bug fix PR is merged we can do another merge PR from 8.0.x to develop

@boubkerbribri would you be ok with that? And @matks if it's accepted could you force merge it please?

**Comment by matks** [comment]
> The failing tests in auto campaign should be fixed by this PR #30326 I think we should still force merge this PR or the merge PR will keep growing week after week, and as soon as the bug fix PR is merged we can do another merge PR from 8.0.x to develop
> 
> @boubkerbribri would you be ok with that? And @matks if it's accepted could you force merge it please?

OK for me if @boubkerbribri is OK about this.

> UI tests code checks / Checking doubles in steps identifiers (pull_request)
Is this OK too?

**Review by NeOMakinG — DISMISSED** [review]
front part LGTM

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    await helper.closeBrowserContext(browserContext);
+  });
+
+  it('should login in BO', async function () {
+    await loginCommon.loginBO(this, page);
+  });
+
+  for (let i = 0; i < numberOfOrderToCreate; i++) {
+    // Pre-condition: Create order in FO
+    createOrderByCustomerTest(orderByCustomerData, `baseContext_${i}`);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
    createOrderByCustomerTest(orderByCustomerData, `${baseContext}_${i}`);
```

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+    // eslint-disable-next-line no-loop-func
+    describe(`Create Credit slip n°${i + 1}`, async () => {
+      it('should go to \'Orders > Orders\' page\'', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `goToOrdersPage${i}`,
+          baseContext,
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
          `${baseContext}_${i}`,
```

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+        const pageTitle = await ordersPage.getPageTitle(page);
+        await expect(pageTitle).to.contains(ordersPage.pageTitle);
+      });
+
+      it('should go to the first order page', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `goToCreatedOrderPage${i}`,
+          baseContext,
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
          `${baseContext}_${i}`,
```

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+        const textMessage = await orderPageProductsBlock.addPartialRefundProduct(page, 1, 1);
+        await expect(textMessage).to.contains(orderPageProductsBlock.partialRefundValidationMessage);
+      });
+
+      it('should check the existence of the Credit slip document', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `checkCreditSlipDocumentName${i}`,
+          baseContext,
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
          `${baseContext}_${i}`,
```

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+        const result = await orderPageTabListBlock.modifyOrderStatus(page, Statuses.shipped.status);
+        await expect(result).to.equal(Statuses.shipped.status);
+      });
+
+      it('should add a partial refund', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `addPartialRefund${i}`,
+          baseContext,
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
          `${baseContext}_${i}`,
```

File: tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+        const pageTitle = await orderPageTabListBlock.getPageTitle(page);
+        await expect(pageTitle).to.contains(orderPageTabListBlock.pageTitle);
+      });
+
+      it(`should change the order status to '${Statuses.shipped.status}' and check it`, async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `updateCreatedOrderStatus${i}`,
+          baseContext,
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/03_creditSlips/02_sortAndPaginationCreditSlips.js]:
```suggestion
          `${baseContext}_${i}`,
```

**Review by matks — CHANGES_REQUESTED** [review]
Argh :( now this PR has conflicts

And as this PR is a merge, you can't rebase so it means the process must be restarted from scratch :(

**Comment by zuk3975** [comment]
> Argh :( now this PR has conflicts
> 
> And as this PR is a merge, you can't rebase so it means the process must be restarted from scratch :(

wow this could get so frustrating :open_mouth: 
How do you guys usually handle such merges? :thinking: (seems like a loop)

**Comment by matks** [comment]
> > Argh :( now this PR has conflicts
> > And as this PR is a merge, you can't rebase so it means the process must be restarted from scratch :(
> 
> wow this could get so frustrating 😮 How do you guys usually handle such merges? 🤔 (seems like a loop)

Any PR merge PR that stays for more than 2 or 3 days is likely to have git conflicts so we try to speed up the process and merge it during a short time window

This PR could not be done fast because UI tests are failing

**Comment by lartist** [comment]
@Progi1984 got rid of the problem which causes the CI failing but unfortunately conflicts had already showed up.

**Comment by lartist** [comment]
This PR can be closed. I opened a new one https://github.com/PrestaShop/PrestaShop/pull/30399

**Comment by zuk3975** [comment]
closing in favour of https://github.com/PrestaShop/PrestaShop/pull/30399

**Pull request closed without merging**